### PR TITLE
Introduced a new type to reference task instances.

### PR DESCRIPTION
**Pull request by tedsteen**
See PR-feedback https://github.com/kagkarlsson/db-scheduler/pull/10#discussion_r112958242.

### CODE REVIEW

File: src/main/java/com/github/kagkarlsson/scheduler/task/TaskInstanceId.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package com.github.kagkarlsson.scheduler.task;
+
+public interface TaskInstanceId {
+    String getTaskName();
+    String getId();
+
+    class StandardTaskInstanceId implements TaskInstanceId {
+        private final String taskName;
```
Comment (kagkarlsson) [inline src/main/java/com/github/kagkarlsson/scheduler/task/TaskInstanceId.java]: Have any idea for how we could get a short-hand for creating a TaskInstanceId? 

Or would you be happy with `new StandardTaskInstanceId(TASK_NAME, ID)` when you are calling reschedule/cancel?

Comment (tedsteen) [inline src/main/java/com/github/kagkarlsson/scheduler/task/TaskInstanceId.java]: How about a static method on `TaskInstanceId`
I have no idea what to name it tho.. `for`, `of`, `new`??
`TaskInstanceId.of("taskName", "id")`

Comment (kagkarlsson) [inline src/main/java/com/github/kagkarlsson/scheduler/task/TaskInstanceId.java]: Sounds good, `TaskInstanceId.of` works :)

Comment (tedsteen) [inline src/main/java/com/github/kagkarlsson/scheduler/task/TaskInstanceId.java]: I learned something new today :) You can create a static method on an interface, but you can only access it from within that interface.
screen shot 2017-05-02 at 10 44 03

Any ideas?

Comment (kagkarlsson) [inline src/main/java/com/github/kagkarlsson/scheduler/task/TaskInstanceId.java]: Is `TaskInstance`an interface here? Or should it have been `TaskInstanceId.of`?

Comment (kagkarlsson) [inline src/main/java/com/github/kagkarlsson/scheduler/task/TaskInstanceId.java]: It should be possible to make `TaskInstanceId` a regular class, and change (on `TaskInstance):
```
private final String id;
```

to 
```
private final TaskInstanceId id;
```

Small duplication of the task-name in both the Task and the TaskInstanceId, but not sure that is bad...

Comment (tedsteen) [inline src/main/java/com/github/kagkarlsson/scheduler/task/TaskInstanceId.java]: >Is TaskInstancean interface here? Or should it have been TaskInstanceId.of?

🤦‍♂️ 
I didn't have my morning coffee or something. It's a static method on the interface now. Don't mind me :)
Commit coming any second now!

**Comment by tedsteen** [comment]
ok @kagkarlsson If you like this the I can merge and it will be part of https://github.com/kagkarlsson/db-scheduler/pull/10 which I think is in a good state now and also ready to be merged.

**Comment by kagkarlsson** [comment]
Looks good, go ahead and merge it

**Pull request merged**